node_cli/core/host.py

```
@@ -65,52 +74,91 @@ def fix_url(url):
 
-    if allocation:
-        update_resource_allocation(env_type)
+    try:
+        logger.info('Preparing host started')
+        make_dirs()
+        save_env_params(env_filepath)
 
+        if allocation:
+            update_resource_allocation(env_type)
+    except Exception as e:
+        error_exit(f'Failed to prepare host: {str(e)}', CLIExitCodes.FAILURE)
 
-def is_node_inited():
-    return os.path.isfile(RESOURCE_ALLOCATION_FILEPATH)
+
+def is_node_inited() -> bool:
+    """Check if the SKALE node has been initialized.
+
+    Determines initialization status by checking for existence of the
+    resource allocation file.
+    """
+    try:
+        # Check if resource allocation file exists as initialization indicator
+        return os.path.isfile(RESOURCE_ALLOCATION_FILEPATH)
+    except OSError as e:
+        logger.error(f'Error checking node initialization status: {e}')
+        return False
 
 
 def make_dirs():
     for dir_path in (
-            SKALE_DIR, NODE_DATA_PATH, CONTAINER_CONFIG_PATH,
-            CONTRACTS_PATH, ETH_STATE_PATH, NODE_CERTS_PATH,
-            REMOVED_CONTAINERS_FOLDER_PATH,
-            SGX_CERTS_PATH, SCHAINS_DATA_PATH, LOG_PATH,
-            REPORTS_PATH, REDIS_DATA_PATH,
-            SKALE_RUN_DIR, SKALE_STATE_DIR, SKALE_TMP_DIR
+        SKALE_DIR,
+        NODE_DATA_PATH,
+        CONTAINER_CONFIG_PATH,
+        CONTRACTS_PATH,
+        ETH_STATE_PATH,
+        NODE_CERTS_PATH,
+        REMOVED_CONTAINERS_FOLDER_PATH,
+        SGX_CERTS_PATH,
+        SCHAINS_DATA_PATH,
+        LOG_PATH,
+        REPORTS_PATH,
+        REDIS_DATA_PATH,
+        SKALE_RUN_DIR,
+        SKALE_STATE_DIR,
+        SKALE_TMP_DIR,
     ):
         safe_mkdir(dir_path)
 
 
-def save_env_params(env_filepath):
-    copyfile(env_filepath, SKALE_DIR_ENV_FILEPATH)
+def save_env_params(env_filepath: str) -> None:
+    """Copy environment parameters file to SKALE directory."""
+    if not os.path.isfile(env_filepath):
```

https://www.reddit.com/r/programming/comments/7xyt5d/overly_defensive_programming/

> If your exception handling makes your code difficult to read, or to have so much boilerplate that people won't read large sections of it, you have destroyed the readability of your code.
Why is readability more important than an obvious bug like swallowing an exception? Bugs can be patched, but code that is hard to read has to be rewritten, which is a much bigger hassle.